packages/itwinui-css/src/input/input.scss

```diff
@@ -100,10 +99,56 @@
   }
 }
 
-.iui-input-with-icon.iui-input-with-icon {
-  @include input-with-icon.iui-input-with-icon;
+// Mixin to have inputs combined with icon (eg. select)
+.iui-input-with-icon {
+  display: grid;
+  align-items: center;
+  cursor: pointer;
+  position: relative;
+
+  > :first-child {
+    grid-area: 1 / -1;
+    padding-inline-end: var(--iui-component-height); // Width of the end-icon button
+  }
+
+  &:focus-within {
+    > :first-child {
+      // --_focus-color can be used to control this from outside
+      outline: 2px solid var(--_focus-color, var(--iui-color-border-accent));
```

Comment:
(unrelated to this PR?) Should we change `--_focus-color` to something like `--_iui-input-with-icon-focus-color`?

Reply:
renamed